### PR TITLE
update prototype

### PR DESCRIPTION
**Pull request by jparisu**
change static functions from DDB to public
Implemented Info classes
Added ShareInfo class

### CODE REVIEW

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ class DiscoveryDataBase
 
     class AckedFunctor
     {
-        using argument_type = eprosima::fastrtps::rtps::ReaderProxy*;
+        using argument_type = eprosima::fastrtps::rtps::ReaderProxy *;
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp:72]:
```suggestion
        using argument_type = eprosima::fastrtps::rtps::ReaderProxy*;
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp [hunk 2]
```diff
@@ -158,78 +157,47 @@ class DiscoveryDataBase
...
 
-    static bool is_writer_(
+    static bool is_writer(
             const eprosima::fastrtps::rtps::CacheChange_t* ch);
 
-    static bool is_reader_(
+    static bool is_reader(
             const eprosima::fastrtps::rtps::CacheChange_t* ch);
 
     static eprosima::fastrtps::rtps::GUID_t guid_from_change_(
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp:190]: Apply on cpp
```suggestion
    static eprosima::fastrtps::rtps::GUID_t guid_from_change(
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantInfo.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (it != writers.end())
+    {
+        writers.erase(it);
+    }
+}
+
+} /* namespace ddb */
+} /* namespace rtps */
+} /* namespace fastdds */
+} /* namespace eprosima */
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantInfo.cpp:73]: In general, blank line at the end of files

File: src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantsAckStatus.hpp [hunk 4]
```diff
@@ -13,24 +13,61 @@
...
+#include <map>
+
+#include <fastdds/rtps/common/GuidPrefix_t.hpp>
+
 namespace eprosima {
 namespace fastdds {
 namespace rtps {
 namespace ddb {
 
+
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantsAckStatus.hpp:32]:
```suggestion
```

File: src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+namespace ddb {
+
+void DiscoverySharedInfo::set_disposal(
+        eprosima::fastrtps::rtps::CacheChange_t* change)
+{
+    relevant_participants_builtin_ack_status_.match_all();
+    change_info(change);
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.cpp:34]: Should we check whether the changes is a disposal?

File: src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.cpp [hunk 6]
```diff
@@ -35,9 +35,8 @@ namespace rtps {
 
 bool EDPServer2::createSEDPEndpoints()
 {
-    // Assert that PDP is a SERVER
-    PDPServer2* pPDP = dynamic_cast<PDPServer2*>(mp_PDP);
-    assert(pPDP);
+    // Assert that there is PDP SERVER
+    assert(mp_PDP);
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.cpp:39]: I think the idea was to assert that the `mp_PDP` is in fact pointing to a `PDPServer2*`
```suggestion
    assert(dynamic_cast<PDPServer2*>(mp_PDP));
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.cpp (hunk 5 above)
Comment (jparisu) [reply src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.cpp:34]: no, that is done in process_data_queue. If there is not a disposal it should not call `set_disposal`

**Review by EduPonz — APPROVED** [review]
LGTM :rocket:

**Pull request merged**